src/Common/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -411,6 +413,38 @@ private static KeyBlobMagicNumber CurveNameToMagicNumber(string name, bool inclu
             }
         }
 
+        /// <summary>
+        /// Map a curve name to magic number. Maps the names of the curves that worked pre-Win10
+        /// to the pre-Win10 magic numbers to support import on pre-Win10 environments 
+        /// that don't have the named curve functionality.
+        /// </summary>
+        private static KeyBlobMagicNumber EcdhCurveNameToMagicNumber(string name, bool includePrivateParameters)
+        {
+            switch (EcdhCurveNameToAlgorithm(name))
+            {
+                case AlgorithmName.ECDHP256:
+                    return includePrivateParameters ?
+                        KeyBlobMagicNumber.BCRYPT_ECDH_PRIVATE_P256_MAGIC :
+                        KeyBlobMagicNumber.BCRYPT_ECDH_PUBLIC_P256_MAGIC;
+
+                case AlgorithmName.ECDHP384:
+                    return includePrivateParameters ?
+                    KeyBlobMagicNumber.BCRYPT_ECDH_PRIVATE_P384_MAGIC :
+                    KeyBlobMagicNumber.BCRYPT_ECDH_PUBLIC_P384_MAGIC;
+
+                case AlgorithmName.ECDHP521:
+                    return includePrivateParameters ?
+                    KeyBlobMagicNumber.BCRYPT_ECDH_PRIVATE_P521_MAGIC :
+                    KeyBlobMagicNumber.BCRYPT_ECDH_PUBLIC_P521_MAGIC;
```

Comment:
Nit: indenting of this and the above case don't match the others

Reply:
Fixed (and the other set)